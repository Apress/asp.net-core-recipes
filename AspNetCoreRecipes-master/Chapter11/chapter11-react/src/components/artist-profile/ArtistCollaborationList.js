import React from 'react';

const ArtistCollaborationList = () => (
  <table className="table table-striped">
    <thead>
      <tr>
        <th>
          Project Name
        </th>
        <th>
          Status
        </th>
        <th>
          Created
        </th>
        <th>
          Modified
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Project 1</td>
        <td>Recruiting / Idea Exchange</td>
        <td>12/28/2016</td>
        <td>1/2/2017</td>
      </tr>
      <tr>
        <td>Project 2</td>
        <td>Mixing</td>
        <td>12/28/2016</td>
        <td>1/2/2017</td>
      </tr>
    </tbody>
  </table>
);

export default ArtistCollaborationList;

