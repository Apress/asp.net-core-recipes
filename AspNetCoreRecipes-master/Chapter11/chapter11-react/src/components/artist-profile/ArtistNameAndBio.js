import React from 'react';
import PropTypes from 'prop-types';

const ArtistNameAndBio = ({artistName, artistBio}) => (
  <div className="artist-profile-text">
    <h1>{artistName.firstName} {artistName.lastName}</h1>
    <p>{artistBio}</p>
  </div>
);

ArtistNameAndBio.propTypes = {
  artistName: PropTypes.shape({
    firstName: PropTypes.string,
    lastName: PropTypes.string
  }).isRequired,
  artistBio: PropTypes.string.isRequired
};
export default ArtistNameAndBio;
