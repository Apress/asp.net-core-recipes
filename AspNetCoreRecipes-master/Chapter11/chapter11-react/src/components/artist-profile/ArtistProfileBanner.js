import React from 'react';
import PropTypes from 'prop-types';

const ArtistProfileBanner = ({ bannerImage }) => (
  <img
    className="artist-profile-banner"
    src={bannerImage}
    alt="Artist profile large background"
  />
);

ArtistProfileBanner.propTypes = {
  bannerImage: PropTypes.string.isRequired
};

export default ArtistProfileBanner;

