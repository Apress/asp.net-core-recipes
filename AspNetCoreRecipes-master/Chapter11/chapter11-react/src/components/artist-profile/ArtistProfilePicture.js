import React from 'react';
import PropTypes from 'prop-types';

const ArtistProfilePicture = ({ profileImage }) => (
  <img
    className="artist-profile-image img-circle img-thumbnail"
    src={profileImage}
    alt="Artist Avatar"
  />
);

ArtistProfilePicture.propTypes = {
  profileImage: PropTypes.string.isRequired
};

export default ArtistProfilePicture;
