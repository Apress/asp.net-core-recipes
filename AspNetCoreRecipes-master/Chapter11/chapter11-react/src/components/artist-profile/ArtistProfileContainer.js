import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ArtistProfileBanner from './ArtistProfileBanner';
import ArtistProfilePicture from './ArtistProfilePicture';
import ArtistNameAndBio from './ArtistNameAndBio';
import ArtistCollaborationList from './ArtistCollaborationList';
import ArtistProfileApi from '../../api/ArtistProfileApi';
import './artistprofile.css';

class ArtistProfileContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      profileImage: props.profileImage,
      bannerImage: props.bannerImage,
      artistName: props.artistName,
      artistBio: props.artistBio
    };
  }

  componentWillMount() {
    ArtistProfileApi.getArtistProfile(1)
      .then((data) => {
        this.setState({
          profileImage: data.profileImageUrl,
          bannerImage: data.bannerImageUrl,
          artistName: { firstName: data.firstName, lastName: data.lastName },
          artistBio: data.bio
        });
      })
      .catch((error) => {
        console.error(`There has been a problem with your fetch operation: ${error.message}`);
      });
  }
  render() {
    return (
      <div className="container-fluid">
        <div className="row">
          <div className="col-lg-12">
            <div className="artist-profile">
              <ArtistProfileBanner bannerImage={this.state.bannerImage} />
              <ArtistProfilePicture profileImage={this.state.profileImage} />
              <ArtistNameAndBio
                artistName={this.state.artistName}
                artistBio={this.state.artistBio}
              />
            </div>
          </div>
        </div>
        <div className="row">
          <div className="col-lg-12">
            <h2>Collaboration Projects</h2>
            <ArtistCollaborationList />
          </div>
        </div>
      </div>
    );
  }
}

ArtistProfileContainer.propTypes = {
  profileImage: PropTypes.string,
  bannerImage: PropTypes.string,
  artistName: PropTypes.shape({
    firstName: PropTypes.string,
    lastName: PropTypes.string
  }),
  artistBio: PropTypes.string
};
ArtistProfileContainer.defaultProps = {
  profileImage: '/images/profileSample.jpg',
  bannerImage: '/images/SampleBackground1.jpg',
  artistName: { firstName: 'firstName', lastname: 'lastname' },
  artistBio: 'Artist Bio. Blaa blaa blaa ...'
};

export default ArtistProfileContainer;
