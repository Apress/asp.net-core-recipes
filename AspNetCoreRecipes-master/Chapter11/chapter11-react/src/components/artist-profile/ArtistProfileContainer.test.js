import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ArtistProfileContainer from './ArtistProfileContainer';
import ArtistProfileBanner from './ArtistProfileBanner';
import ArtistProfilePicture from './ArtistProfilePicture';
import ArtistNameAndBio from './ArtistNameAndBio';
import ArtistCollaborationList from './ArtistCollaborationList';


describe('ArtistProfileContainer', () => {
  it('ArtistProfileContainer Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ArtistProfileContainer />, div);
  });

  it('ArtistProfileContainer should render Banner, Picture, CollaborationList and NameAndBio', () => {
    const wrapper = shallow(<ArtistProfileContainer />);
    expect(wrapper.contains([
      <ArtistProfilePicture />
    ])).toBe(true);

    expect(wrapper.contains([
      <ArtistProfileBanner />
    ])).toBe(true);

    expect(wrapper.contains([
      <ArtistNameAndBio />
    ])).toBe(true);

    expect(wrapper.contains([
      <ArtistCollaborationList />
    ])).toBe(true);
  });


  it('ArtistProfileContainer Matches Snapshot', () => {
    const component = renderer.create(<ArtistProfileContainer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
