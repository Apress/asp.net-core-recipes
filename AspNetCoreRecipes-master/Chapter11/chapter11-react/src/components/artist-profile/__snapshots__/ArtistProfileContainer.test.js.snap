// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistProfileContainer ArtistProfileContainer Matches Snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <div
        className="artist-profile"
      >
        <img
          alt="Artist profile large background"
          className="artist-profile-banner"
          src="/images/SampleBackground1.jpg"
        />
        <img
          alt="Artist Avatar"
          className="artist-profile-image img-circle img-thumbnail"
          src="/images/profileSample.jpg"
        />
        <div
          className="artist-profile-text"
        >
          <h1>
            ArtistFirst ArtistLast
          </h1>
          <p>
            Artist Bio. Blaa blaa blaa ...
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <h2>
        Collaboration Projects
      </h2>
      <table
        className="table table-striped"
      >
        <thead>
          <tr>
            <th>
              Project Name
            </th>
            <th>
              Status
            </th>
            <th>
              Created
            </th>
            <th>
              Modified
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Project 1
            </td>
            <td>
              Recruiting / Idea Exchange
            </td>
            <td>
              12/28/2016
            </td>
            <td>
              1/2/2017
            </td>
          </tr>
          <tr>
            <td>
              Project 2
            </td>
            <td>
              Mixing
            </td>
            <td>
              12/28/2016
            </td>
            <td>
              1/2/2017
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
